hazelcast/src/main/java/com/hazelcast/instance/impl/HazelcastBootstrap.java

```diff
@@ -182,6 +183,15 @@ public static synchronized void executeJar(@Nonnull Supplier<HazelcastInstance>
         }
     }
 
+    private static void checkHazelcastCodebasePresence(JarFile jarFile) {
+        List<String> classFiles = JarScanner.findClassFiles(jarFile, HazelcastBootstrap.class.getSimpleName());
```

Comment:
I think it makes more sense to use the canonical name matching to understand the presence (for example, in the test you added even if there is no hazelcast code presence in the jar, the presence of the same named simple class causes a warning). I don't think that the name of a user class would be `HazelcastBootstrap`, so I think using the simple name for matching is good enough for this case, but I would prefer to use canonicalName under normal conditions (I'm aware that it complicates the algorithm for scanning the jar). Maybe we can add a comment that we don't expect such a user class.

Reply:
I've decided to use `HazelcastBootstrap` as a marker since it's quite unique (I've checked GitHub repositories and there's a single duplicate that seems to be not used with hz-cli): https://github.com/search?q=HazelcastBootstrap+filename%3AHazelcastBootstrap.java+-org%3Ahazelcast&type=Code). 

Using something internal would make this test more fragile. `HazelcastBootstrap` is the main entry point so I've assumed it will be stable for a long time.

I would consider keeping it easy as at the end of the day, it's only a warning so in case of unlikely problems it won't break anything and we will easily adjust it. What do you think?